pkg/kubectl/resource/builder.go

```diff
@@ -539,17 +547,21 @@ func (b *Builder) visitorResult() *Result {
 		if err != nil {
 			return &Result{err: err}
 		}
-
 		visitors := []Visitor{}
 		for _, mapping := range mappings {
 			client, err := b.mapper.ClientForMapping(mapping)
 			if err != nil {
 				return &Result{err: err}
 			}
+
 			selectorNamespace := b.namespace
 			if mapping.Scope.Name() != meta.RESTScopeNameNamespace {
 				selectorNamespace = ""
 			}
+			if err := b.validateNamespace(client, selectorNamespace); err != nil {
```

Comment:
I would have expected this to be in the error path - if a Get fails, check any specified namespaces and report those along with the error.  We would accumulate selectorNamespaces and then use that during error analysis.  And make the visitors responsible for this check.